crypto/bn/bn_rand.c

```diff
@@ -111,23 +99,33 @@ static int bnrand(BNRAND_FLAG flag, BIGNUM *rnd, int bits, int top, int bottom)
     return 0;
 }
 
+int BN_rand_ex(BIGNUM *rnd, int bits, int top, int bottom, BN_CTX *ctx)
+{
+    return bnrand(NORMAL, rnd, bits, top, bottom, ctx);
+}
 int BN_rand(BIGNUM *rnd, int bits, int top, int bottom)
 {
-    return bnrand(NORMAL, rnd, bits, top, bottom);
+    return BN_rand_ex(rnd, bits, top, bottom, NULL);
```

Comment:
Since the actual implementation is in the private helper function named `bnrand()`, I would prefer if both public implementations `BN_rand()` and `BN_rand_ex()` would be based on it, instead of basing the implementation of the former on the latter

```C
int BN_rand(BIGNUM *rnd, int bits, int top, int bottom)
{
   return bnrand(NORMAL, rnd, bits, top, bottom, NULL);
}
```